epyqlib/device.py

```diff
@@ -1216,8 +1220,8 @@ def absolute_path(self, path=""):
     def get_frames(self):
         return self.frames
 
-    @pyqtSlot(bool)
-    def bus_status_changed(self, online, transmit):
+    @pyqtSlot(bool, bool, bool)
```

Comment:
Good catch of the existing error.  I generally don't bother with the decorator at all anymore.  I think it has some relevance to really hot slots and when connecting across threads (I think we only do this to receive from python-can).

Reply:
I probably wouldn't have touched this if I wasn't adding a parameter. I too have noticed that these `pyqtSlot` decorators tend to be optional. I have this vague memory of actually needing to add the decorator to some methods in a past application or maybe I had the type parameter(s) incorrect and it caused an issue. That was a few years ago. It's probably good practice and certainly the decorator should have the correct types if it is defined at all.

Reply:
Anymore, most of my objects aren't even inheriting from `QObject` so `@pyqtSlot` can't be used afaik.  For signals I use https://github.com/altendky/stlib/blob/6b75b2493aa94eb7eaeece0c38a6bc472533e3a4/epyqlib/utils/qt.py#L1181-L1219 (actually documented at https://github.com/altendky/qtrio/blob/5de075813fe73dd2a411556dbf338cd24917892c/qtrio/_qt.py#L12-L68).